tests/ignore-invalid-enum-values.ts

```diff
@@ -0,0 +1,107 @@
+import createTSURL, {
+  optionalEnumArray,
+  optionalEnum,
+  requiredEnumArray,
+  requiredEnum,
+} from '../src';
+
+enum TestEnum {
+  ONE = 'ONE',
+  TWO = 'TWO',
+  THREE = 'THREE',
+}
+
+describe('ignoreInvalidEnums', () => {
+  it('should allow deconstructing a URL or path that has invalid enum values without throwing', () => {
+    const url1 = createTSURL(
+      [
+        'test',
+        requiredEnum('test1', TestEnum),
+        optionalEnum('test2', TestEnum),
+      ],
+      {
+        baseURL: 'https://example.com',
+        basePath: '/api',
+        queryParams: [],
+      }
+    );
+
+    expect(() =>
+      url1.deconstruct('https://example.com/api/test/FOUR/THREE')
+    ).toThrow(/Invalid value for part "test1" - FOUR/);
+    expect(
+      url1.deconstruct('https://example.com/api/test/FOUR/THREE', {
+        ignoreInvalidEnums: true,
+      })
+    ).toEqual({
+      urlParams: {
+        test1: 'FOUR',
+        test2: 'THREE',
+      },
+      queryParams: {},
+    });
```

Comment:
We shouldn't return invalid enum values, as the types won't match the values. This function should throw due to required `test1` not having a valid value.